src/App.tsx

```diff
@@ -17,18 +98,34 @@
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onStatusChange={handleStatusChange}
+                onSearchChange={handleSearchChange}
+                onClearSearch={handleClearSearch}
+                searchQuery={searchQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading && <Loader />}
+
+              <TodoList
+                todos={getFilteredTodos()}
+                onSelectTodo={handleSelectTodo}
+                selectedTodoId={selectedTodo?.id}
+              />
```

Comment:
Verify that the `TodoList` component accepts the props `todos`, `onSelectTodo`, and `selectedTodoId` as expected. Ensure these props are correctly typed and implemented in the `TodoList` component.